### ISSUE TITLE
Import job fails after being successfully paused.

### ISSUE THREAD

**Issue by plaflamme** [cockroachdb/cockroach#36900]
**Describe the problem**

Paused job fails after being successfully paused.

**To Reproduce**

Unclear if this can reproduced reliably yet.

**Expected behavior**

Pausing a job shouldn't succeed until all related processing is actually paused.

**Additional data / screenshots**

The reported failure was:

```
internal error: uncaught error: cannot update progress on paused job (id 443635913143812106)
gs://plaflamme-inbox-poc/part-00116-da5737c0-1b52-44fa-ac04-a4741d960b54-c000.csv.gz
github.com/cockroachdb/cockroach/pkg/ccl/importccl.readInputFiles.func1
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:141
github.com/cockroachdb/cockroach/pkg/ccl/importccl.readInputFiles
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:149
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*csvInputReader).readFiles
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_csv.go:80
github.com/cockroachdb/cockroach/pkg/ccl/importccl.(*readImportDataProcessor).doRun.func1
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/read_import_proc.go:488
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:170
github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1333
```

**Environment:**
 - CockroachDB version: `v19.1.0-rc.2`
 - Server OS: `GKE`
 - Client app: `cockroach sql`

**Additional context**

The `IMPORT` job was using `experimental_direct_ingestion`. It is a fairly large import of 127 files between 0.5 and 9GB each.

Seems related to https://github.com/cockroachdb/cockroach/issues/30505 but this has a specific error message provided.

**Comment by vivekmenezes** [comment]
@plaflamme thanks for reporting this problem. I suspect a fix for it could make it into release 19.2.

**Comment by dt** [comment]
Closing as a duplicate of https://github.com/cockroachdb/cockroach/issues/30505 but we still haven't made any progress on this one.
